python/tests/test_user_session.py

```diff
@@ -0,0 +1,93 @@
+import base64
+import json
+from unittest import mock
+
+import pytest
+
+from numerous._client._graphql_client import GraphQLClient
+from numerous.generated.graphql.client import Client as GQLClient
+from numerous.user_session import Session
+
+
+class UserStub:
+    def __init__(self, user_info: dict[str, str], client: GraphQLClient) -> None:
+        self.id = user_info["user_id"]
+        self.name = user_info["name"]
+        self._client = client
+
+    @staticmethod
+    def from_user_info(user_info: dict[str, str], _client: GraphQLClient) -> "UserStub":
+        """Create a MockUser instance from a user info dictionary."""
+        return UserStub(user_info, _client)
```

Comment:
We can just use the `User` type directly instead of this stub.

If you wanted to create this, it would be simpler to define it as a `dataclass` or even `NamedTuple` instead. Then you get the constructor for free.

And also the `_client` is not used here?

Reply:
Changed to use User directly for type checking